art-decisions/proposals/2022-10-26-standardize-python-formatting.md

```diff
@@ -0,0 +1,74 @@
+# Title
+
+Standardize on python code formatting requirements
+
+All Python code must conform to [PEP 8](https://peps.python.org/pep-0008), and
+will be checked via the use of [`flake8`](https://github.com/PyCQA/flake8).
+
+All Python imports will be required to be sorted and grouped according to [PEP
+8](https://peps.python.org/pep-0008/#imports) using
+[`isort`](https://github.com/PyCQA/isort).
+
+All Python code (Python 3 assumed) will be required to be auto-formatted via
+[`black`](https://github.com/psf/black).
+
+A pre-commit hook will be offered for use to help verify the above
+requirements before they go through the CI job.
+
+The CI environment will include a series of checks to ensure the above
+requirements are met using the following commands:
+
+```
+$ isort --profile black --check .
+$ black --check .
+$ flake8 .
+```
+
+Acceptance of this proposal would require applying the formatting first to all
+existing code at the same time as adding the `pre-commit` hooks.
+
+## Recommendation
+
+We should set one standard and adhere to it judiciously without exception. I
+recommend that we simply adopt the standards already established by the
+[pbench](https://github.com/distributed-system-analysis/pbench) repository and
+eliminate further debate on the topic. This formatting standard should also be
+applied wherever possible with automation and CI.
```

Comment:
Based on Janos's feedback, it may make sense to reword it to something like:
"It is important that we adhere to one standard judiciously and eliminate further debate on the topic. The standards documented above match the standards already adopted by the [pbench](https://github.com/distributed-system-analysis/pbench) repository. This formatting standard should also be applied wherever possible with automation and CI.

Reply:
Was this comment addressed?

Reply:
nope, not yet.

Reply:
update pushed